src/tests/end-to-end/tests/details-view/tabstop.test.ts

```diff
@@ -0,0 +1,104 @@
+describe('Tabstop tests', () => {
+    describe('tabstop from fastpass', () => {
+        let browser: Browser;
+        let targetPage: TargetPage;
+        let detailsViewpage: Page;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetPage = await browser.newTargetPage();
+            detailsViewpage = await browser.newDetailsViewPage(targetPage);
+
+            await goToTabStopTest();
+            await enableToggleByAriaLabel();
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+            }
+        });
+
+        test('if visualHelper is turned on after starting tabstop and pressing tabs on target page', async () => {
+            await targetPage.bringToFront();
+            await targetPage.keyPress('Tab');
+
+            const nativeWidgetHeadingTargetPage = await targetPage.getSelectorElement(TargetPageElementSelectors.targetPageNativeWidgetH1);
+            await nativeWidgetHeadingTargetPage.click();
+
+            const frame = targetPage.frames().find(f => f.name() === TargetPageElementSelectors.targetPageNativeWidgetIFrameName);
+            const button = await frame.$(TargetPageElementSelectors.targetPageNativeWidgetFirstRadio);
+            await button.click();
+
+            const shadowRoot = await targetPage.getShadowRoot();
+            await targetPage.waitForDescendentSelector(shadowRoot, fastPassSelectors.tabStopVisulizationStart, { visible: true });
+
+            for (let i = 0; i < 3; i++) {
+                await targetPage.keyPress('Tab');
+            }
+
+            expect(await targetPage.getShadowRootHtmlSnapshot()).toMatchSnapshot();
+        });
+
+        async function goToTabStopTest(): Promise<void> {
+            await detailsViewpage.clickSelector(fastPassSelectors.tabStopNavButtonSelector);
+        }
+
+        async function enableToggleByAriaLabel(): Promise<void> {
+            await detailsViewpage.clickSelector(fastPassSelectors.tabStopToggleUncheckedSelector);
+
+            await detailsViewpage.waitForSelector(fastPassSelectors.tabStopToggleCheckedSelector);
+        }
+    });
+
+    describe('tabstop from adhoc tools', () => {
+        let browser: Browser;
+        let targetPage: TargetPage;
+        let popupPage: PopupPage;
+
+        beforeAll(async () => {
+            browser = await launchBrowser({ suppressFirstTimeDialog: true });
+            targetPage = await browser.newTargetPage();
+            popupPage = await browser.newPopupPage(targetPage);
+            await popupPage.gotoAdhocPanel();
+        });
+
+        afterAll(async () => {
+            if (browser) {
+                await browser.close();
+            }
+        });
+
+        test('matches snapshot when tabstop is used from adhoc panel', async () => {
+            await popupPage.enableToggleByAriaLabel('Tab stops');
+
+            await targetPage.bringToFront();
```

Comment:
Were you able to determine why explicitly `bringToFront()`ing and clicking the elements on the page were necessary? These are steps I would expect should not be required by this test, so if we really need to keep them in, it would be nice to see a comment explaining why (ideally referencing an issue)